sdk/eventhub/azure-eventhub/azure/eventhub/_common.py

```diff
@@ -266,24 +292,26 @@ def system_properties(self):
 
         if self._sys_properties is None:
             self._sys_properties = {}
-            if self.message.properties:
+            if self._raw_amqp_message.properties:
                 for key, prop_name in _SYS_PROP_KEYS_TO_MSG_PROPERTIES:
-                    value = getattr(self.message.properties, prop_name, None)
+                    value = getattr(self._raw_amqp_message.properties, prop_name, None)
                     if value:
                         self._sys_properties[key] = value
-            self._sys_properties.update(self.message.annotations)
+            self._sys_properties.update(self._raw_amqp_message.annotations)
         return self._sys_properties
 
     @property
     def body(self):
-        # type: () -> Union[bytes, Iterable[bytes]]
-        """The content of the event.
-
-        :rtype: bytes or Generator[bytes]
+        # type: () -> Any
+        """The body of the Message. The format may vary depending on the body type:
+        For ~azure.eventhub.AmqpMessageBodyType.DATA, the body could be bytes or Iterable[bytes]
+        For ~azure.eventhub.AmqpMessageBodyType.SEQUENCE, the body could be List or Iterable[List]
+        For ~azure.eventhub.AmqpMessageBodyType.VALUE, the body could be any type.
+        :rtype: Any
         """
         try:
-            return self.message.get_data()
-        except TypeError:
+            return self._raw_amqp_message.body
+        except:
             raise ValueError("Event content empty.")
 
     def body_as_str(self, encoding="UTF-8"):
```

Comment:
could you check `body_as_str` and `body_as_json` behavior for body of data of multiple sections (e.g. sending a list of strings)/sequence/value?